src/stratigraphy/util/geometric_line_utilities.py

```diff
@@ -235,7 +235,7 @@ def merge_parallel_lines(lines: list[Line], tol: int = 8, angle_threshold: float
         return merged_lines
 
 
-def _odr_regression(x: ArrayLike, y: ArrayLike) -> tuple:
+def _odr_regression(x: ArrayLike, y: ArrayLike, weights: ArrayLike) -> tuple:
```

Comment:
Would be good to add a default value for weights. Something like:
`weights: ArrayLike = None`

And then something like
```
if weights is None:
    weights = np.ones((len(x), ))
```